Only load the script if doNotTrack is not set

Jivings opened the pull request:
Hey! This PR means that the SA script will only be injected into the DOM if the `doNotTrack` option is not set.

How it works is that instead of added the script tag that loads `hello.js`, we use inline JavaScript to create and inject the script tag only if DNT is false.

- I used the DNT checking code from [here](https://dev.to/corbindavenport/how-to-correctly-check-for-do-not-track-with-javascript-135d).
- I removed the `defer` option from the script tag, because it's now deferred anyway

My editors prettier config reformatted everything, so if you don't want that then it would be great if you could add a `.prettierrc` file with your config to the repo and I'll merge and reformat again.

adriaanvanrossum commented on src/gatsby-ssr.js:
```diff
@@ -1,23 +1,61 @@
+      m.doNotTrack == "yes" ||
+      m.doNotTrack == "1" ||
+      m.msDoNotTrack == "1" ||
+      (s.external.msTrackingProtectionEnabled && s.external.msTrackingProtectionEnabled())
```
This is deprecated, I would suggest to remove it.

https://developer.mozilla.org/en-US/docs/Web/API/Window/external

Jivings replied:
Depends on browser support I suppose. This line would be for <IE9 I think? Which your probably safe not supporting but please confirm :)

adriaanvanrossum replied:
I'm very happy to drop support for <IE8, if you are still using that you have serious issues and privacy shouldn't be your biggest priority. Security is more important in that case I would say.

adriaanvanrossum commented on src/gatsby-ssr.js:
```diff
@@ -1,23 +1,61 @@
+// 4. Inserts the script tag before the first other script in the document
+const loadScript = (domain, options) => {
+  return `!(function(s, i, m, p, l, e) {
+    if (
```
We could do something like this:

```js
var dnt = s.doNotTrack || m.doNotTrack || m.msDoNotTrack
if (['1', 'yes'].indexOf(dnt) > -1) return console.warn('Simple Analytics: Not loading script when doNotTrack is enabled');
```

Jivings replied:
I assumed the weak equality check on the article I mentioned was intentional. Maybe in some browsers the 1 is a string and others a number? In which case this might not work... Any idea?

adriaanvanrossum replied:
From the spec it's not super clear: https://www.w3.org/TR/tracking-dnt/#expression-format but on MDN it's pretty clear: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/doNotTrack

There seem to be no exceptions to browsers using a string or not. They all use a string with either `yes` or `1`. https://caniuse.com/#search=dnt

I would keep the strong equality check because that's the only way browsers do it.

<img width="337" alt="CleanShot 2019-07-17 at 12 44 15@2x" src="https://user-images.githubusercontent.com/1079135/61369582-b364de80-a890-11e9-9429-e244f09e3c36.png">

adriaanvanrossum commented on src/gatsby-ssr.js:
```diff
@@ -1,23 +1,61 @@
+// 3. Sets all the options properties on the script tag
+// 4. Inserts the script tag before the first other script in the document
+const loadScript = (domain, options) => {
+  return `!(function(s, i, m, p, l, e) {
```
Nice touch

Jivings commented:
Added .prettierrc file and updated the code style.

Jivings commented:
Sorry for delay. Updated with your feedback. Changed DNT code to this;

```
const dnt = s.doNotTrack || m.doNotTrack || m.msDoNotTrack;
if (/yes|1/.test(dnt)) {
  return console.warn('Simple Analytics: Not loading script when doNotTrack is enabled');
}
```

<img width="173" alt="Screen Shot 2019-08-18 at 12 08 36" src="https://user-images.githubusercontent.com/1462828/63220122-98e59400-c1b3-11e9-8676-028c9b161cf6.png">
